elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLConversionUtils.java

```diff
@@ -133,19 +152,49 @@ public  GraphQLList classToQueryMap(Class<?> keyClazz,
                                         Class<?> valueClazz,
                                         DataFetcher fetcher,
                                         ConversionLedger ledger) {
-        return new GraphQLList(
+        String mapName = toValidNameName(keyClazz.getName() + valueClazz.getCanonicalName() + MAP);
+
+        if (mapConversions.containsKey(mapName)) {
+            return mapConversions.get(mapName);
+        }
+
+        GraphQLOutputType keyType;
+        if (keyClazz.isEnum()) {
+            keyType = classToEnumType(keyClazz);
+        } else {
+            keyType = fetchScalarOrObjectOutput(keyClazz, ledger, fetcher);
+            if (keyType == null) {
```

Comment:
This if statement will never be hit.  fetchScalarOrObjectOutput already does this check.